Assets/Scripts/Animation/PopupInventoryItem.cs

```diff
@@ -1,105 +1,73 @@
 ﻿using System.Collections;
-using System.Collections.Generic;
 using UnityEngine;
 using UnityEngine.UI;
 
 public class PopupInventoryItem : MonoBehaviour {
-    private static Image gameObjectImage;
-    private static Vector3 originalPos;
-    private static bool itemRendered = false;
+    private Image gameObjectImage;
+    private Vector3 originalPos;
+    
     [SerializeField]
-    private float bounceMagnitude;
+    private float bounceSpeed = 1.5f;
     [SerializeField]
-    private float maxHeight;
+    private float bounceHeight = 0.5f;
+    [SerializeField]
+    private int bounceRepetitions = 2;
 
-    void Start() {
+    private void Start() {
         gameObjectImage = gameObject.GetComponent<Image>();
-        originalPos = new Vector3(gameObjectImage.transform.position.x, gameObjectImage.transform.position.y, gameObjectImage.transform.position.z);
-        SetImageActive(false);
-        bounceMagnitude = 1.0f;
-        maxHeight = 5.0f;
+        originalPos = transform.position;
+        gameObject.SetActive(false);
     }
 
-    /// <summary>
-    /// attach the prefab's sprite to current object's sprite in image component and render it
-    /// </summary>
-    public static void renderItemSprite(InventoryItem prefab) {
-        Image prefabImage = prefab.GetComponent<Image>();
-        SetImageSprite(prefabImage.sprite);
-        SetImageActive(true);
-        itemRendered = true;
+    private void OnEnable() {
+        StartCoroutine(AnimateSprite());
     }
 
-    private void Update() {
-        if (itemRendered) { StartCoroutine(AnimateSprite()); } 
+    /// <summary>
+    /// Attach the prefab's sprite to current object's sprite in image component and render it
+    /// </summary>
+    public void RenderItemSprite(InventoryItem prefab) {
+        gameObjectImage.sprite = prefab.GetComponent<Image>().sprite;
+        gameObject.SetActive(true);
     }
 
    private IEnumerator GoUp() {
-        float currHeight = gameObjectImage.transform.position.y;
-        while (currHeight < maxHeight) {
-            gameObjectImage.transform.Translate(0.0f, bounceMagnitude * Time.deltaTime, 0.0f);
-            currHeight = gameObjectImage.transform.position.y;
+        while (transform.position.y < originalPos.y + bounceHeight) {
```

Comment:
It's important that the comparison here is relative to the original position, in case we need to move things around.